src/plugins/intel_cpu/src/graph.h

```diff
@@ -253,9 +254,24 @@ class Graph {
     friend std::shared_ptr<ngraph::Function> dump_graph_as_ie_ngraph_net(const Graph &graph);
 
 private:
+    struct OutputNodesMap {
+        void InsertNode(const NodePtr&, const std::unordered_set<std::string>&);
+        const std::unordered_map<NodePtr, std::unordered_set<std::string>>& get() const {
+            return nodesMap;
+        }
+        NodePtr getNodePtrByName(const std::string& name) const;
+        bool hasOutputWithName(const std::string& name) const;
+        void clear() {
+            nodesMap.clear();
+        }
+
+        private:
+            std::unordered_map<NodePtr, std::unordered_set<std::string>> nodesMap;
+    };
```

Comment:
1. May be it is a matter of taste, but once the struct has a constructor and a private field, it should be converted to a class. It simply does not correspond to a C struct anymore and has all the features of a typical class.
2. All the requests are made by `name`, so we probably should have a map where `name` is a key. Why can't we just use name->ptr map as before? Having multiple names should not be a problem as long as they map the same pointer, shouldn't it? Actually this is a main question. So far I do not see significant advantages of having `std::unordered_map<NodePtr, std::unordered_set<std::string>>`

Reply:
1. Modified
2. The new approach provides the following benefits:
- Eliminates repetitive actions on a single node when iterating by name;
- Prevents cases when node was recreated with additional names in tensor;